haystack_argilla/base.py

```diff
@@ -21,30 +28,89 @@ def __init__(
-            ## TODO: Add other warnings/errors
-        
-        ## TODO: Add checks for SDK and Server versions
-        ## TODO: Add checks to confirm ApiKey and ApiUrl are valid
+            self.ARGILLA_VERSION = rg.__version__
+
+        except ImportError:
+            raise ImportError(
+                "To use the Argilla callback manager you need to have the `argilla` "
+                "Python package installed. Please install it with `pip install argilla`"
+            )
+
+        ## Check whether the Argilla version is compatible
+        if parse(self.ARGILLA_VERSION) < parse("1.20.0"):
+            raise ImportError(
+                f"The installed `argilla` version is {self.ARGILLA_VERSION} but "
+                "`ArgillaCallbackHandler` requires at least version 1.21.0. Please "
+                "upgrade `argilla` with `pip install --upgrade argilla`."
+            )
         
-        # Connect to Argilla with the given credentials
+        # API_URL and API_KEY
+        # Show a warning message if Argilla will assume the default values will be used
+        if api_url is None and os.getenv("ARGILLA_API_URL") is None:
+            warnings.warn(
+                (
+                    "Since `api_url` is None, and the env var `ARGILLA_API_URL` is not"
+                    f" set, it will default to `{self.DEFAULT_API_URL}`, which is the"
+                    " default API URL in Argilla Quickstart."
+                ),
+            )
+            api_url = self.DEFAULT_API_URL
+
+        if api_key is None and os.getenv("ARGILLA_API_KEY") is None:
+            warnings.warn(
+                (
+                    "Since `api_key` is None, and the env var `ARGILLA_API_KEY` is not"
+                    f" set, it will default to `{self.DEFAULT_API_KEY}`, which is the"
+                    " default API key in Argilla Quickstart."
+                ),
+            )
+            api_key = self.DEFAULT_API_KEY
+
+        ## TODO: Add checks for Server versions
+    
+        # Connect to Argilla with the provided credentials, if applicable
         try:
             rg.init(
-                api_key=api_key,
-                api_url=api_url,
+                api_key=api_key, 
+                api_url=api_url
+            )
+        except Exception as e:
+            raise ConnectionError(
+                f"Could not connect to Argilla with exception: '{e}'.\n"
+                "Please check your `api_key` and `api_url`, and make sure that "
+                "the Argilla server is up and running. If the problem persists "
+                f"please report it to {self.ISSUES_URL} as an `integration` issue."
+            ) from e
+        
+        # Set the Argilla variables
+        self.dataset_name = dataset_name
+        self.workspace_name = workspace_name or rg.get_workspace()
+        
+        # Retrieve the `FeedbackDataset` from Argilla
+        try:
```

Comment:
if the dataset is not present, I would expect to create a dataset with required format.